token-lending/program/tests/helpers/mod.rs

```diff
@@ -508,7 +508,7 @@ impl TestLendingMarket {
             obligation,
         } = args;
 
-        let dex_market_orders_pubkey = if deposit_reserve.dex_market.is_some() {
+        let dex_market_orders_pubkey = if deposit_reserve.dex_market.is_none() {
```

Comment:
I can see why this got confusing -- I've had to recreate the mental model a lot of times going over this!  This seems correct though